sklearn/pipeline.py

```diff
@@ -803,10 +813,5 @@ def make_union(*transformers):
                               n_components=2, n_iter=5,
                               random_state=None, tol=0.0))],
            transformer_weights=None)
-
-
-    Returns
-    -------
-    f : FeatureUnion
     """
-    return FeatureUnion(_name_estimators(transformers))
+    return FeatureUnion(_name_estimators(transformers), **kwargs)
```

Comment:
I like the idea of being conservative, @jnothman

Maybe something like the following (but perhaps with a KeyError instead)?

```python
def make_union(*transformers, **kwargs):
    """Construct a FeatureUnion from the given transformers.

    This is a shorthand for the FeatureUnion constructor; it does not require,
    and does not permit, naming the transformers. Instead, they will be given
    names automatically based on their types. It also does not allow weighting.

    Parameters
    ----------
    *transformers : list of estimators

    n_jobs : int, optional
        Number of jobs to run in parallel (default 1).

    Returns
    -------
    f : FeatureUnion

    Examples
    --------
    >>> from sklearn.decomposition import PCA, TruncatedSVD
    >>> from sklearn.pipeline import make_union
    >>> make_union(PCA(), TruncatedSVD())    # doctest: +NORMALIZE_WHITESPACE
    FeatureUnion(n_jobs=1,
           transformer_list=[('pca',
                              PCA(copy=True, iterated_power='auto',
                                  n_components=None, random_state=None,
                                  svd_solver='auto', tol=0.0, whiten=False)),
                             ('truncatedsvd',
                              TruncatedSVD(algorithm='randomized',
                              n_components=2, n_iter=5,
                              random_state=None, tol=0.0))],
           transformer_weights=None)
    """
    if 'n_jobs' in kwargs:
        return FeatureUnion(_name_estimators(transformers), **kwargs)
    elif kwargs:
        raise NotImplementedError
    else:
        return FeatureUnion(_name_estimators(transformers))
```

This gives us the following output:
```bash
In [9]: make_union(PCA(), TruncatedSVD())
Out[9]: 
FeatureUnion(n_jobs=1,
       transformer_list=[('pca', PCA(copy=True, iterated_power='auto', n_components=None, random_state=None,
  svd_solver='auto', tol=0.0, whiten=False)), ('truncatedsvd', TruncatedSVD(algorithm='randomized', n_components=2, n_iter=5,
       random_state=None, tol=0.0))],
       transformer_weights=None)

In [10]: make_union(PCA(), TruncatedSVD(), n_jobs=4)
Out[10]: 
FeatureUnion(n_jobs=4,
       transformer_list=[('pca', PCA(copy=True, iterated_power='auto', n_components=None, random_state=None,
  svd_solver='auto', tol=0.0, whiten=False)), ('truncatedsvd', TruncatedSVD(algorithm='randomized', n_components=2, n_iter=5,
       random_state=None, tol=0.0))],
       transformer_weights=None)

In [11]: make_union(PCA(), TruncatedSVD(), anyOtherArg=3)
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-11-87bdeace7aa2> in <module>()
----> 1 make_union(PCA(), TruncatedSVD(), anyOtherArg=3)

/Users/acb/Documents/DataSci/contrib/scikit/scikit-learn/sklearn/pipeline.py in make_union(*transformers, **kwargs)
    818         return FeatureUnion(_name_estimators(transformers), **kwargs)
    819     elif kwargs:
--> 820         raise NotImplementedError
    821     else:
    822         return FeatureUnion(_name_estimators(transformers))

NotImplementedError: 
```

Thank you both for all your input on this 👍